### ISSUE TITLE
Ambiguous use of 'flatten', new in 4.2.x

### ISSUE THREAD

**Issue by nickvelloff** [ReactiveCocoa/ReactiveCocoa#2960]
Perhaps related to https://github.com/ReactiveCocoa/ReactiveCocoa/issues/2530

This worked before updating to `4.2.x`.

I'm new to ReactiveCocoa and have no idea how to resolve this. 

``` Swift
var signalProducers: [SignalProducer<Void, NSError>] = [ some0SignalProducer ]
signalProducers += some1SignalProducers
signalProducers += some2SignalProducers
signalProducers += some3lSignalProducers
            
return SignalProducer(values: signalProducers)
            .flatten(FlattenStrategy.Concat)
```

**Comment by mdiep** [comment]
You'll need to declare the type of the signal producer. We added overloads that handle `NoError`, but that can confuse the type inferencer in some cases.

This should work:

```swift
var signalProducers: [SignalProducer<Void, NSError>] = [ some0SignalProducer ]
signalProducers += some1SignalProducers
signalProducers += some2SignalProducers
signalProducers += some3lSignalProducers

return SignalProducer<SignalProducer<Void, NSError>, NoError>(values: signalProducers)
            .flatten(FlattenStrategy.Concat)
```

Sorry for the disruption!

**Comment by nickvelloff** [comment]
@mdiep saved me! 🙇

**Comment by nikita-leonov** [comment]
@mdiep is it something that planned to be addressed in future? We got a lot of explicit types now because of this. It is not so fast to write it now, as well as it is not concise and harder to read.

**Comment by ikesyo** [comment]
IIUC this should be already addressed by https://github.com/ReactiveCocoa/ReactiveCocoa/pull/3043 and released as v4.2.2.

**Comment by nikita-leonov** [comment]
@ikesyo thanks! I noticed 4.2.2 is not released in CocoaPods yet. Is it something that core contributors can handle? I see you listed in owners for a pod.

**Comment by NachoSoto** [comment]
Which it still needs a test, as mentioned in #3043 😁

**Comment by nikita-leonov** [comment]
@NachoSoto Yeap it is missing unit tests, but already merged into 4.2.2 and already released :( It seems logical to have 4.2.2 released in CocoaPods too.

**Comment by ikesyo** [comment]
@NachoSoto The test is already added in #3048. :wink:

**Comment by NachoSoto** [comment]
@ikesyo oh derp of course. Thanks!

**Comment by nikita-leonov** [comment]
@ikesyo ok I handled it. @ashfurrow shared access to podspec trunk and I released 4.2.2.
